src/conn/conn_compact.c

```diff
@@ -145,55 +145,63 @@ __background_compact_should_run(WT_SESSION_IMPL *session, const char *uri, int64
 }
 
 /*
- * __background_compact_start --
+ * __wt_background_compact_start --
  *     Pre-fill compact related statistics for the given file.
  */
-static int
-__background_compact_start(
-  WT_SESSION_IMPL *session, const char *uri, int64_t id, WT_BACKGROUND_COMPACT_STAT **compact_stat)
+int
+__wt_background_compact_start(WT_SESSION_IMPL *session)
 {
-    WT_BACKGROUND_COMPACT_STAT *temp_compact_stat;
+    WT_BACKGROUND_COMPACT_STAT *compact_stat;
     WT_BM *bm;
     WT_DECL_RET;
+    int64_t id;
+    const char *uri;
 
     bm = S2BT(session)->bm;
+    id = S2BT(session)->id;
+    uri = bm->block->name;
 
-    temp_compact_stat = __background_compact_get_stat(session, uri, id);
+    compact_stat = __background_compact_get_stat(session, uri, id);
 
     /* If the table is not in the list, allocate a new entry and insert it. */
-    if (temp_compact_stat == NULL) {
-        WT_ERR(__wt_calloc_one(session, &temp_compact_stat));
-        WT_ERR(__wt_strdup(session, uri, &temp_compact_stat->uri));
-        temp_compact_stat->id = (uint32_t)id;
-        __background_compact_list_insert(session, temp_compact_stat);
+    if (compact_stat == NULL) {
+        WT_ERR(__wt_calloc_one(session, &compact_stat));
+        WT_ERR(__wt_strdup(session, uri, &compact_stat->uri));
+        compact_stat->id = (uint32_t)id;
```

Comment:
Do we need the cast here?

Reply:
Good point, not anymore

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/pull/9541/commits/bc23e3a1385951c7f233cd0e5a2b55b70266edc0.